### PR TITLE
Added default value option to ConfigurationListElement

### PR DESCRIPTION
**Pull request by mkrebs81**
Modified the XML schema to accept default values.
Also updated the ABB B23 specification with configuration parameters.

Requires modification of SGrJava implementation in order to work.

### CODE REVIEW

**Comment by mkrebs81** [comment]
also added serial port name to Modbus RTU parameters

File: SchemaDatabase/SGr/Product/Product.xsd [hunk 1]
```diff
@@ -27,6 +27,7 @@ https://github.com/SmartgridReady/SGrSpecifications  -->
     <sequence>
       <element name="name" type="string" />
       <element name="dataType" type="sgr:DataTypeProduct" />
+      <element name="defaultValue" type="string" />
```
Comment (ArndGerns) [inline SchemaDatabase/SGr/Product/Product.xsd:30]: I'ld make the defaultValue optional with `minOccurs="0"`. Then you don't need to define it everywhere.

**Comment by ergo-furrer** [comment]
> also added serial port name to Modbus RTU parameters

Hi @mkrebs81. Thanks for that!  

I recently encountered the same problem while implementing the SGr-Intermediary  in Java since it was not possible to handle the COM-Port the same way as the other properties.

**Review by ArndGerns — CHANGES_REQUESTED** [review]

**Comment by mkrebs81** [comment]
> Hi @ergo-furrer, @mkrebs81,
> 
> If we add the serial port, then some of the EID's are not valid any more.
> 
> Is there always a COM port? Then we need to update the existing EID's.
> 
> And what is that in `mvpEi4Modbus_0000_0036_0030_4_0.1_SGCP_BiDirFlexMgmt.xml`:
> 
> ```
>       <modbusInterfaceDescription>
>         <modbusInterfaceSelection>TCPIP</modbusInterfaceSelection>
>         <modbusTcp>
>           <port>512</port>
>           <address>192.168.1.155</address>
>           <slaveId>0</slaveId>
>         </modbusTcp>
>         <modbusRtu>
>           <slaveAddr>10</slaveAddr>
>           <baudRateSelected>19200</baudRateSelected>
> ```
> 
> There the port is in the TCPIP part? Can it really be, that both RTU and TCP are there ?

I think it is possible that a Modbus RTU device either uses RTU over serial port _OR_ RTU over a TCP/IP gateway (RTU over TCP/IP is not the same as Modbus TCP!)

I distinguish them by checking whether the tcp address or serial port name are empty, with preference to serial.

**Comment by ergo-furrer** [comment]
> I distinguish them by checking whether the tcp address or serial port name are empty, with preference to serial.

Its looks like a viable solution to me. 

It means in the EI-XML above at least `<address>` and  `<comPort>` must property placeholders.

> <modbusInterfaceDescription>
>         <modbusInterfaceSelection>TCPIP</modbusInterfaceSelection>
>         <modbusTcp>
>           <port>512</port>
>           <address>{{ip_address}}</address>
>           <slaveId>0</slaveId>
>         </modbusTcp>
>         <modbusRtu>
>           <comPort>{{com_port}}</comPort>
>           <slaveAddr>10</slaveAddr>
>           <baudRateSelected>19200</baudRateSelected>

**Comment by mkrebs81** [comment]
Thanks for the feedback.

> How many options are there? Three options:
> 
> O1) RTU with serial port O2) TCP with IP and port O3) RTU over TCP with IP and port for TCP - do the serial options really make sense here?

IMHO these are the three options that may occur in a real scenario.
Of course O3 does not need any serial options. It can be configured like O2.

**Comment by ergo-furrer** [comment]
> Thanks for the feedback.
> 
> > How many options are there? Three options:
> > O1) RTU with serial port O2) TCP with IP and port O3) RTU over TCP with IP and port for TCP - do the serial options really make sense here?
> 
> IMHO these are the three options that may occur in a real scenario. Of course O3 does not need any serial options. It can be configured like O2.

I agree, from the EI-XML point of view O1 and O2 are sufficient.

**Comment by ergo-furrer** [comment]
@mkrebs8 Just for your information. I am currently working on this branch too to fix the EI-XML's. Some of them do not validate anymore due to the changes. I will come back when I am finished.

**Review by ArndGerns — APPROVED** [review]

**Review by ArndGerns — DISMISSED** [review]

**Review by ergo-furrer — APPROVED** [review]

**Review by ergo-furrer — APPROVED** [review]

**Pull request merged**